ale/teacher/hybrid/sequential_representation_lc.py

```diff
@@ -0,0 +1,181 @@
+from typing import List, Any, Dict
+import random
+import numpy as np
+from gensim.models import Word2Vec,KeyedVectors
+from sklearn.cluster import KMeans
+from ale.config import NLPTask
+from ale.corpus.corpus import Corpus
+from ale.registry.registerable_teacher import TeacherRegistry
+from ale.teacher.base_teacher import BaseTeacher
+from ale.trainer.predictor import Predictor
+from ale.teacher.exploration.k_means import ClusteredDocuments, ClusterDocument
+from ale.trainer.prediction_result import TokenConfidence, PredictionResult
+
+class NGramVectors:
+    def __init__(self, sizes: List[int], seed: int, lexical_dimension: int = 50, word_embedding_dimension: int = 50) -> None:
+        self.sizes: List[int] = sizes
+        self.seed: int = seed
+        self.lexical_dimension: int = lexical_dimension
+        self.word_embedding_dimension: int = word_embedding_dimension
+        self.vector_dict: Dict[str,np.ndarray] = dict()
+
+    def generate_n_grams(self, token: str) -> List[np.ndarray]:
+        """ Generates n grams of the given sizes
+        """ 
+        subsequences: List[str] = []
+        vectors: List[np.ndarray] = []
+        for size in self.sizes:
+            subsequences_of_size: List[str] = [token[i:i+size] for i in range(len(token)-size)]
+            subsequences.extend(subsequences_of_size)
+        for seq in subsequences:
+            seq_vector = self.vector_dict.get(seq)
+            if not seq_vector: # vector n gram does not exist yet
+                np.random.seed(self.seed) # use ALE seed
+                self.vector_dict[seq] = np.random.random(size=(self.lexical_dimension)) # lexical vector of dimension given, per default 50
+            vectors.append(self.vector_dict[seq])
+        return vectors
```

Comment:
Do we need the vectors at this point? The first step would be to create a mapping of each n-gram to lexical vector?